Exclude dependencies from license checks (#423)
actions/dependency-review-action: README.md

| `allow-ghsas`                   | Contains a list of GitHub Advisory Database IDs that can be skipped during detection.                                                                                             | Any GHSAs from the [GitHub Advisory Database](https://github.com/advisories) | none          |
| `license-check`                 | Enable or disable the license check performed by the action.                                                                                                                      | `true`, `false`                                                              | `true`        |
| `vulnerability-check`           | Enable or disable the vulnerability check performed by the action.                                                                                                                | `true`, `false`                                                              | `true`        |

> | `allow-dependencies-licenses`\* | Containts excluded packages per ecosystem for the licenses checks.                                                                                                                | Any package(s) in [purl](https://github.com/package-url/purl-spec) format    | none          |

Reviewer: ```suggestion
| `allow-dependencies-licenses`\* | Contains a list of packages that will be excluded from license checks.                                                                                                                | Any package(s) in [purl](https://github.com/package-url/purl-spec) format    | none          |
```

| `allow-ghsas`                     | Contains a list of GitHub Advisory Database IDs that can be skipped during detection.                                                                                             | Any GHSAs from the [GitHub Advisory Database](https://github.com/advisories) | none          |
| `license-check`                   | Enable or disable the license check performed by the action.                                                                                                                      | `true`, `false`                                                              | `true`        |
| `vulnerability-check`             | Enable or disable the vulnerability check performed by the action.                                                                                                                | `true`, `false`                                                              | `true`        |

> | `allow-dependencies-licenses`\*\* | Containts excluded packages per ecosystem for the licenses checks.                                                                                                                | `ecosystemName: pkgName1, pkgName2`                                          | none          |

Reviewer 2: ```suggestion
| `allow-dependencies-licenses`\*\* | Contains excluded packages per ecosystem for the licenses checks.                                                                                                                | `ecosystemName: pkgName1, pkgName2`                                          | none          |
```

  - 'GPL-3.0'
  - 'BSD-3-Clause'
  - 'MIT'
allow-dependencies-licenses:
  npm: @actions/core, @actions/github

>   pip: requests

Author: I see what you mean. I added it to the example here for a similar reason as my previous comment in the options table... Since it was mentioning `allow-licenses`, I though it would be a good spot to exemplify the options to exclude packages. I will remove it for now as per your suggestion.

Would creating a `docs/examples.md` or `docs/advanced-usage.md` be an option? We could link to it from this README to more configuration examples and possible uses of this action?

>   pip: requests

Reviewer 2: > We could link to it from this README to more configuration examples and possible uses of this action?

I love this idea, it'd allow us to show multiple examples of more advanced usage.

>   pip: requests

Author: I've added a file in [1896d6f](https://github.com/actions/dependency-review-action/pull/423/commits/1896d6f93643a7d934aa99e1a335e8e5a3199e27) and put in first few examples that came to mind. 

I guess we will have to restructure as hopefully the examples grow but I think this works for now.

```

### Considerations

- Checking for licenses is not supported on Enterprise Server.
- The action will only accept one of the two `license` parameters; an error will be raised if you provide both.
- We don't have license information for all of your dependents. If we can't detect the license for a dependency **we will inform you, but the action won't fail**.

> - The `allow-dependencies-licenses` option can only be used with the external configuration file.

Author: I kept this only to be possible via the `config-file` for two reasons:
- Configuring this inline would make the workflow file get more complicated and I tried to avoid that.
- If I try to pass an object in Actions on that level in the same format I get a syntax error that an `object` is not allows on that level. Having different formats for the same option when configuring it inline and config sounded like a bad idea. 😄 
```
      - name: 'Dependency Review'
        uses: theztefan/dependency-review-action@allow-list-dependencies
        with:
          fail-on-severity: 'critical'
          allow-licenses: MIT, BSD-3-Clause
          comment-summary-in-pr: true
          config-file: thez-org/open-book/dependancy-review/org-wide.yml@main
          allow-dependencies-licenses:                  <--- "Invalid type found: one of string , number , boolean were expected but an object was found"
            npm: "@actions/core, @actions/github"
            pip: "requests
```
Happy to revisit this if you have a suggestion.

> - The `allow-dependencies-licenses` option can only be used with the external configuration file.

Reviewer 2: It's a shame we can't use maps in workflow files.

Something that your example brought to my attention is that we don't have a list of possible ecosystems, and I don't want to be maintaining one every time the Dependency Graph adds a new ecosystem.

I think that using [purl](https://github.com/package-url/purl-spec) is the way to go here, I see a few advantages:
1. It's already being supported by multiple other software supply chain initiatives (including SPDX, which is what we use for licenses).
2. It'd allow us to keep supporting the same configuration both inline and externally.
3. Someone else takes care of defining what the ecosystem keys are.

An inline copy of your example would look like this:

```
allow-dependencies-licenses: "pkg:npm/@actions/core, pkg:npm/@actions/github, pkg:pip/requests"
```

I guess the external one would be:

```
allow-dependencies-licenses:
  - 'pkg:npm/@actions/core'
  - 'pkg:npm/@actions/github'
  - 'pkg:pip/requests'
```

What do you think?

> - The `allow-dependencies-licenses` option can only be used with the external configuration file.

Author: Love the idea! I switched it to `purl` format. So, it supports both config file and inline configuration.
